### ISSUE TITLE
Upload doesn't work after page refresh

### ISSUE THREAD

**Issue by KasparScherrer** [vaadin/vaadin-upload-flow#147]
Using Vaadin 14.0.0

I have an Upload Component inside a View that is annotated with `@PreserveOnRefresh`.  

My Upload Component works in the first page load, but as soon as I refresh the page it will no longer work: 
![upload forbidden](https://i.imgur.com/qfLSDC5.png)  
console error:
![upload console error](https://i.imgur.com/I95aqAZ.png)

```
@Route(value = "Test")
@PreserveOnRefresh
public class TestView extends VerticalLayout {

    public TestView(){
        add(new H4("Test View"));
        addUpload();
    }

    private void addUpload() {
        Upload upload = new Upload(new MultiFileMemoryBuffer());
        upload.setUploadButton(new Button("Test Upload", VaadinIcon.UPLOAD.create()));
        upload.setDropAllowed(false);
        
        int i = 0;
        upload.addSucceededListener(event -> {
            Notification.show(String.format("Upload #%d succeeded. Filename: '%s'", i, event.getFileName()));
        });
        upload.addFailedListener(event -> {
            Notification.show(String.format("Upload #%d failed. Filename: %s", i, event.getFileName()));
        });
        upload.addFinishedListener(event -> i++);
        add(upload);
    }
}
```

**Comment by web-padawan** [comment]
See also code to reproduce here: https://github.com/vaadin/vaadin-upload-flow/issues/154#issue-516994046

**Comment by web-padawan** [comment]
Related to vaadin/flow#6478

**Comment by web-padawan** [comment]
Suggested workaround:

```
    @Override
    protected void onAttach(AttachEvent e) {
        getElement().setAttribute("target", new StreamReceiver(
                getElement().getNode(), "upload", getStreamVariable()));
    }
```

**Comment by KasparScherrer** [comment]
I cannot do that by extending the Upload class, because `getStreamVariable()` has private access. Can you please show me how to implement this workaround?

**Comment by web-padawan** [comment]
I think we should implement that in the component. Will discuss with a team.

**Comment by dschulz** [comment]
I just tested and this bug is still reproducible in `vaadin-upload-flow:2.1.0`

**Comment by KasparScherrer** [comment]
This issue still blocks me from using `@PreserveOnRefresh` completely...

**Comment by tomivirkki** [comment]
This is the same as https://github.com/vaadin/flow/issues/6478

The issue was fixed by https://github.com/vaadin/flow/pull/7627 but it seems that the fix hasn't been backported to V14.X

**Comment by tomivirkki** [comment]
Related issue for cherry-picking the fix: https://github.com/vaadin/flow/issues/8625

**Comment by alvarezguille** [comment]
Fix was cherry-picked and included in vaadin 14.2.3 and 14.3.0. Closing this issue
